src/setfit/onnx.py

```diff
@@ -0,0 +1,227 @@
+from .modeling import SetFitModel
+from sentence_transformers import models
+import numpy as np
+import onnx
+import warnings
+from typing import Optional, Union
+import torch
+from sklearn.linear_model import LogisticRegression
+
+
+class ONNXSetFitModel(torch.nn.Module):
+    def __init__(
+        self,
+        model_body,
+        pooler,
+        model_head: Optional[Union[torch.nn.Module, LogisticRegression]] = None,
+    ):
+        super().__init__()
+
+        self.model_body = model_body
+        self.pooler = pooler
+        self.model_head = model_head
+
+    def forward(self, input_ids, attention_mask, token_type_ids):
+        hidden = self.model_body(input_ids, attention_mask, token_type_ids)
+        hidden = {"token_embeddings": hidden[0], "attention_mask": attention_mask}
+
+        hidden = self.pooler(hidden)["sentence_embedding"]
+        if self.model_head is None:
```

Comment:
Should we allow a user to export a model without a head? Just wondering whether to raise a `ValueError` here

Reply:
Sorry, this is confusing.  This OnnxSetFitModel is a wrapper that makes it easy to use torch.onnx.export to export the model_body with the pooling.  In the case of the torch based head we can use this to do the whole onnx export. However when we have an sklearn head we need to export the model_body with the pooler and then the sklearn head.  This was my attempt to make it easy to do just two onnx exports but then it means that the class name doesn't fully represent what is happening here.

Reply:
Thanks, this makes sense!